src/main/java/core/mate/academy/service/MachineServiceImpl.java

```
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        }
+        if (Excavator.class.equals(type)) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        }
+        if (Track.class.equals(type)) {
+            return new ArrayList<>(new TrackProducer().get());
+        }
+        return null;
```

return `null` isn't a good practice
let's throw an exception or return Collections.emptyList()

---

fixed